virtcontainers/qemu_ppc64le.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2018 IBM
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+
+	govmmQemu "github.com/intel/govmm/qemu"
+)
+
+type qemuPPC64le struct {
+	// inherit from qemuArchBase, overwrite methods if needed
+	qemuArchBase
+}
+
+const defaultQemuPath = "/usr/libexec/qemu-kvm"
+
+const defaultQemuMachineType = QemuPPC64le
+
+const defaultQemuMachineOptions = "accel=kvm,usb=off"
+
+const defaultPCBridgeBus = "pci.0"
+
+var qemuPaths = map[string]string{
+	QemuPPC64le: defaultQemuPath,
+}
+
+var kernelParams = []Param{
+	{"tsc", "reliable"},
+	{"no_timer_check", ""},
+	{"rcupdate.rcu_expedited", "1"},
+	{"i8042.direct", "1"},
+	{"i8042.dumbkbd", "1"},
+	{"i8042.nopnp", "1"},
+	{"i8042.noaux", "1"},
+	{"noreplace-smp", ""},
+	{"reboot", "k"},
+	{"console", "hvc0"},
+	{"console", "hvc1"},
+	{"iommu", "off"},
+	{"cryptomgr.notests", ""},
+	{"net.ifnames", "0"},
+	{"pci", "lastbus=0"},
```

Comment:
this improves boot time of PCIe machines types, does QEMU ppc have PCIe machine types?

Reply:
@devimc : Thanks for pointing out. ```pci=lastbus=0``` is an intel only feature. POWER does not use it. So we can remove this option for ppc64le.
Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/admin-guide/kernel-parameters.txt#n3058
```{"iommu", "off"}``` can also be removed. 
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/admin-guide/kernel-parameters.txt#n1699
``` {"cryptomgr.notests", ""},``` and ``` {"net.ifnames", "0"},```  should improve the boot times.

Reply:
Thanks for the parameters analysis, this was a very complete answer :+1:

Reply:
@jodh-intel : Is it fine to raise a PR to optimize the kernel parameters for ppc64le.

Reply:
Hi @nitkon - Sure - please do! ;)